hawkular-alerts-engine/src/main/java/org/hawkular/alerts/engine/cache/CacheManager.java

```diff
@@ -82,64 +85,91 @@ public void init() {
                 publishCache.clear();
             }
             msgLog.infoInitPublishCache();
-            updateActiveIds();
+            requestCacheUpdate();
+
             definitions.registerListener(e -> {
-                updateActiveIds();
-            }, CONDITION_CHANGE, TRIGGER_REMOVE);
+                requestCacheUpdate();
+            }, TRIGGER_CONDITION_CHANGE, TRIGGER_REMOVE);
+
         } else {
             msgLog.warnDisabledPublishCache();
         }
     }
 
-    private synchronized void updateActiveIds() {
+    // Just run updateCache one time if multiple requests come in while an update is already in progress...
+    private void requestCacheUpdate() {
+        log.debug("Cache update requested");
+        if (updateRequested) {
+            log.debug("Cache update, redundant request ignored.");
+            return;
+        }
+
+        updateRequested = true;
```

Comment:
I think it is a good idea to declare volatile on both 'updateRequested' and 'updating', to keep the most recent values in memory at all times. I will change it.